source/v4/saml.yaml

```diff
@@ -0,0 +1,182 @@
+        '401':
+          $ref: '#/responses/Unauthorized'
+        '403':
+          $ref: '#/responses/Forbidden'
+        '501':
+          $ref: '#/responses/NotImplemented'
+
+  /saml/certificate/public:
+    post:
+      tags:
+        - SAML
+      summary: Upload public certificate
+      description: |
+        Upload the public certificate to be used with your SAML configuration. This will also set the filename for the PublicCertificateFile setting in your `config.json`.
+        ##### Permissions
+        Must have `manage_system` permission.
+      parameters:
+        - name: certificate
+          in: formData
+          description: The public certificate file
+          required: true
+          type: file
+      responses:
+        '200':
+          description: SAML certificate upload successful
+          schema:
+            $ref: '#/definitions/StatusOK'
+        '400':
+          $ref: '#/responses/BadRequest'
+        '401':
+          $ref: '#/responses/Unauthorized'
+        '403':
+          $ref: '#/responses/Forbidden'
+        '501':
+          $ref: '#/responses/NotImplemented'
+
+    delete:
+      tags:
+        - SAML
+      summary: Remove public certificate
+      description: |
+        Delete the current public certificate being used with your SAML configuration. This will also disable SAML on your system as this certificate is required for SAML.
+        ##### Permissions
+        Must have `manage_system` permission.
+      responses:
+        '200':
+          description: SAML certificate delete successful
+          schema:
+            $ref: '#/definitions/StatusOK'
+        '401':
+          $ref: '#/responses/Unauthorized'
+        '403':
+          $ref: '#/responses/Forbidden'
+        '501':
+          $ref: '#/responses/NotImplemented'
+
+  /saml/certificate/private:
+    post:
+      tags:
+        - SAML
+      summary: Upload private key
+      description: |
+        Upload the private key to be used with your SAML configuration. This will also set the filename for the PrivateKeyFile setting in your `config.json`.
+        ##### Permissions
+        Must have `manage_system` permission.
+      parameters:
+        - name: certificate
+          in: formData
+          description: The private key file
+          required: true
+          type: file
+      responses:
+        '200':
+          description: SAML certificate upload successful
+          schema:
+            $ref: '#/definitions/StatusOK'
+        '400':
+          $ref: '#/responses/BadRequest'
+        '401':
+          $ref: '#/responses/Unauthorized'
+        '403':
+          $ref: '#/responses/Forbidden'
+        '501':
+          $ref: '#/responses/NotImplemented'
+
+    delete:
+      tags:
+        - SAML
+      summary: Remove private key
+      description: |
+        Delete the current private key being used with your SAML configuration. This will also disable SAML on your system as this key is required for SAML.
```

Comment:
as before this one is used to decrypt the SAML assertions so you can actually have SAML configured without this, the only thing is that if you enable encryption you will need the public certificate too